### PR TITLE
Added Commit to GitRepository

### LINKED ISSUE #441: Add Commit to GitRepository
### Description
Nuke.Common.Git.GitRepository interacts with the local repository and parses the HEAD file.

All that is needed is reading the contents of .git\refs\heads\XXXX and exposing it in a Commit property on the GitRepository type

Edit.
In the event that the HEAD is detached the id will be located in HEAD file 

### Alternatives
Use GitVersion - But this approach seems overkill as this is information already is available

### PR DESCRIPTION
**Pull request by lars-nielsen**
Fixes #441

Added Commit property to GitRepository + ability to be injected from CI-systems

I confirm that the pull-request:

- [x] Follows the contribution guidelines
- [x] Is based on my own work
- [x] Is in compliance with my employer

### CODE REVIEW

File: source/Nuke.Common/Git/GitRepository.cs [hunk 1]
```diff
@@ -51,12 +45,53 @@ public static GitRepository FromLocalDirectory(string directory, string branch =
...
+            // Don't parse commit if it has been supplied from outer scope
+            if (commit == null)
+            {
+                // Attached, follow the ref for the commit
+                if (attachedHead.Success)
+                {
+                    var headRefFile = Path.Combine(gitDirectory, attachedHead.Groups["reference"].Value);
+                    ControlFlow.Assert(File.Exists(headRefFile), $"File.Exists({headRefFile})");
+
+                    commit = File.ReadAllLines(headFile).First();
```
Comment (matkoch) [inline source/Nuke.Common/Git/GitRepository.cs]: `headRefFile`?

Comment (lars-nielsen) [reply source/Nuke.Common/Git/GitRepository.cs]: Couldn't figure out a better name..

It is the file being referenced by HEAD file

(.git/HEAD)
ref: refs/heads/gitrepo-commit

(.git\refs\heads\gitrepo-commit)
0460fa91473c6334d012caf81b69c42bd35103ec

Comment (matkoch) [reply source/Nuke.Common/Git/GitRepository.cs]: That’s not what I mean. Currently Head and Commit both read from the same file?

Comment (lars-nielsen) [reply source/Nuke.Common/Git/GitRepository.cs]: Indeed. And they shouldn't - Fixed and pushed

**Pull request merged**